### DISCUSSION TITLE
Dynamically switching between databases.

### DISCUSSION THREAD

**Discussion by timnewsham** [go-gorm/gorm#5564] [Q&A]
I would like to build GORM queries that use a single database pool but dynamically select between different databases running on the same database server (postgresql in this case). Looking around I found a few solutions:

- Use GORM's DB resolver to select between several statically declared connections.  I can't use this because I will have a large number of dynamically created databases and they should all share the same connection to the same server.
- Use a transaction to manually `use DBNAME`. This seems to require that I give up a lot of the benefits of using GORM to construct my database accesses.
- Use the `Table("db_name.table_name")` construct to explicitly pick the database name and database table for each database access.  This is much better, but I have to give up the automatic table name selection features of GORM.

Is there an easier way to do this that would let me specify just the db_name but allow GORM to still dynamically pick the table_name in the normal way (ie. based off of the model object that is being accessed)? 

Ideally I would be able to use the chainable API to add a constraint that fixes the db_name without explicitly choosing the table name.

Tim

**Comment by jinzhu** [comment]
Hi @timnewsham 

Maybe you can use https://gorm.io/docs/scopes.html#Dynamically-Table

**Reply by Valdenirmezadri** [reply]
same problem here, I need to change the database because I will do a multi tenant.  each customer is a separate database.  I still haven't figured out how to solve it

**Reply by aiph11** [reply]
Hi @Valdenirmezadri, did you already solved for that case? coz i have the same problem with you for each tenant is a separated database.

**Reply by Valdenirmezadri** [reply]
> Comentário original em inglês - Traduzir para português

hi @aiph11 , I solved the problem by creating a connection for each tenant/database

**Comment by robertobelli** [comment]
> Hi @timnewsham
> 
> Maybe you can use https://gorm.io/docs/scopes.html#Dynamically-Table

Hi @jinzhu , but this solution will fix the Preload tables? I have a problem trying a similar method, that fix the name of the "principal" table but not the tables queried by the use of preload. There is a way to change the tablename dynamically of the preload table?
